### ISSUE TITLE
Distinguish Log messages from task output.

### ISSUE THREAD

**Issue by copumpkin** [sbt/sbt#1414]
There's a lot of noise in info messages from a regular sbt run. Dependency resolution is particularly noisy, so I turn my logLevel up to only show me warnings.

But then when I run the test task, I most certainly do want to see the output. Unfortunately, test results and summaries are also output using the info log level.

Ultimate goal is just to have a good signal-to-noise ratio.

**Comment by jsuereth** [comment]
I agree about signal to noise, but I think the premise is at odds with our login design (and redesign coming).

Primarily, `last` and `export` commands should show you all output associated with a task, therefore stdout, stderr and "log" should all feed through the same event-logger in a recallable way.

IN addition, the default noise level should be low.   I haven't quite gone the full marathon on hacking STDOUT for Ivy resolution, but we're halfway there in terms of turning pure noisy STDOUT into something that can be squelched and recalled.

In any case, the title is a definite won't fix.  I'm titling as a feature request for "Less noise in default update logs", as I think this is what the primary issue you're reporting is, correct?

**Comment by copumpkin** [comment]
No, my primary issue (I don't think I wrote this title?) is that given the amount of normal noise, I turn log level up, but that turns off things that I do care about.

It seems fairly "inherent" to running a test suite to want to know what happened, so it seems to merit a first-class output mechanism that isn't just logInfo. Same if I run another task that inherently is supposed to give me feedback: I don't want that going through the standard logging framework, because I don't want adjusted log levels to hide intended output.

The fact that the default build is noisy is a separate concern, but I can deal with that.

**Comment by copumpkin** [comment]
To clarify, some processes (in any program) produce useful human-readable output as they perform their primary duties. Some of it might be debug info, other times you might get warning messages. Those things should go through the logging framework.

Other processes are actually _intended_ to give the operator feedback, as their primary purpose. Take `show` for example. If I type `show someKey` in, I expect it to show me something regardless of log level. It's not an incidental "by the way, here's some useful output" kind of deal. Same (as I see it) with the test summary output from `test`.  Perhaps individual test successes along the way are info-level log messages that would be suppressed by a raised log level, but ultimately I asked for a summary and I'd like to see one independent of external factors.

So I guess what I'm asking for is, in a sentence: distinguish handy log messages from intended output.

**Comment by frgomes** [comment]
I'd like to suggest that logging messages and all other informational messages are redirected to _stderr_, not _stdout_. The main reason lies behind the ability of writing Scala Scripts, which would send useful output to _stdout_.

**Comment by jeffrey-aguilera** [comment]
`sbt` is WAY too noisy.  A simple call to `sbt about` can print hundreds of lines of meaningless information.

**Comment by mgladdish** [comment]
Really annoying that this is still open.

I don't want a CI log full of 

`[info] Resolving blah blah blah` 

but I most definitely do want to see a log full of 

```
[info] MySpec:
[info] scenario 1
...
[info] All tests passed.
```

**Comment by losvald** [comment]
IMHO, logging to `stderr` certainly makes sense in the case of `StdoutOutput` as `outputStrategy`; otherwise, how can you, e.g., run a main class and isolate only the output produced by the program, without the SBT noise?
